src/jsMain/kotlin/io/beatmaps/playlist/playlistTable.kt

```diff
@@ -0,0 +1,193 @@
+                visItem = 0
+                visPage = 0
+                visiblePages = visPage.until(visPage + totalVisiblePages)
+            }
+
+            window.setTimeout(::loadNextPage, 0)
+        }
+    }
+
+    private fun getUrl(page: Int) =
+        if (props.userId != null) {
+            "${Config.apibase}/playlists/user/${props.userId}/$page"
+        } else {
+            props.search?.let { search ->
+                "${Config.apibase}/playlists/search/$page?sortOrder=${search.sortOrder}" +
+                    (if (search.curated != null) "&curated=${search.curated}" else "") +
+                    (if (search.verified != null) "&verified=${search.verified}" else "") +
+                    (if (search.search.isNotBlank()) "&q=${encodeURIComponent(search.search)}" else "") +
+                    (if (search.maxNps != null) "&maxNps=${search.maxNps}" else "") +
+                    (if (search.minNps != null) "&minNps=${search.minNps}" else "") +
+                    (if (search.from != null) "&from=${search.from}" else "") +
+                    (if (search.to != null) "&to=${search.to}" else "") +
+                    (if (search.includeEmpty != null) "&to=${search.includeEmpty}" else "")
+            } ?: ""
+        }
+
+    private fun loadNextPage() {
+        if (state.loading == true)
+            return
+
+        val toLoad = state.visiblePages.firstOrNull { !state.pages.containsKey(it) } ?: return
+
+        setState {
+            loading = true
+        }
+
+        Axios.get<PlaylistSearchResponse>(
+            getUrl(toLoad),
+            generateConfig<String, PlaylistSearchResponse>()
+        ).then {
+            setState {
+                loading = false
+                pages = pages.plus(toLoad to it.data.docs)
+            }
+            window.onscroll = ::handleScroll
+            if (it.data.docs.isNotEmpty()) {
+                window.setTimeout(::handleScroll, 1)
+            }
+        }.catch {
+            // Oh noes
+        }
+    }
+
+    @Suppress("UNUSED_PARAMETER")
+    private fun handleScroll(e: Event) {
+        val scrollPosition = window.pageYOffset
+        val windowSize = window.innerHeight
+
+        val item = ((scrollPosition - 19) / rowHeight).toInt()
+        if (item != state.visItem) {
+            val totalVisiblePages = ceil(windowSize / pageHeight).toInt()
+            setState {
+                visItem = item
+                visPage = item / playlistsPerPage
+                visiblePages = visPage.until(visPage + totalVisiblePages)
+            }
+            props.updateScrollIndex?.invoke(item + 1)
+        }
+
+        loadNextPage()
+    }
+
+    private fun lastPage() = max(state.visiblePages.last, state.pages.maxByOrNull { it.key }?.key ?: 0)
+
+    override fun RBuilder.render() {
+        if (props.visible == false) return
+
+        if (props.own == true) {
+            routeLink("/playlists/new", className = "btn btn-success btn-sm") {
```

Comment:
I don't really have an answer but this button looks lost now. Either needs to be bigger, have some dialog around it or.... something

Image for people so they don't have to get this running to see:
![image](https://user-images.githubusercontent.com/534069/187086439-69b60b81-a661-4ac1-b48e-6d76076424a7.png)